docs/proposals/scheduling/reschedule-task/reschedule-task.md

```diff
@@ -0,0 +1,261 @@
+---
+title: Introduce a mechanism to actively triggle rescheduling
+authors:
+  - "@chaosi-zju"
+reviewers:
+  - "@RainbowMango"
+  - "@chaunceyjiang"
+  - "TBD"
+approvers:
+  - "@RainbowMango"
+  - "TBD"
+
+creation-date: 2024-01-30
+---
+
+# Introduce a mechanism to actively trigger rescheduling
+
+## Background
+
+According to the current implementation, after the replicas of workload is scheduled, it will remain inertia and the 
+replicas distribution will not change. 
+
+However, in some scenarios, users hope to have means to actively trigger rescheduling.
+
+### Motivation
+
+Assuming the user has propagated the workloads to member clusters, replicas migrated due to member cluster failure.
+
+However, the user expects an approach to trigger rescheduling after member cluster restored, so that replicas can
+migrate back.
+
+### Goals
+
+Introduce a mechanism to actively trigger rescheduling of workload resource.
+
+### Applicable scenario
+
+This feature might help in a scenario where: the `replicas` in resource template or `placement` in policy has not changed, 
+but the user wants to actively trigger rescheduling of replicas.
+
+## Proposal
+
+### Overview
+
+This proposal aims to introduce a mechanism of active triggering rescheduling, which benefits a lot in application 
+failover scenarios. This can be realized by introducing a new API, and a new field would be marked when this new API 
+called, so that scheduler can perceive the need for rescheduling.
+
+### User story
+
+In application failover scenarios, replicas migrated from primary cluster to backup cluster when primary cluster failue.
+
+As a user, I want to trigger replicas migrating back when cluster restored, so that:
+
+1. restore the disaster recovery mode to ensure the reliability and stability of the cluster.
+2. save the cost of the backup cluster.
+
+### Notes/Constraints/Caveats
+
+This ability is limited to triggering rescheduling. The scheduling result will be recalculated according to the
+Placement in the current ResourceBinding, and the scheduling result is not guaranteed to be exactly the same as before
+the cluster failure.
+
+> Notes: pay attention to the recalculation is basing on Placement in the current `ResourceBinding`, not "Policy". So if
+> your activation preference of Policy is `Lazy`, the rescheduling is still basing on previous `ResourceBinding` even if
+> the current Policy has been changed.
+
+## Design Details
+
+### API change
+
+* Introduce a new API named `Reschedule` into a new apiGroup `command.karmada.io`:
+
+```go
+//revive:disable:exported
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Reschedule represents the desire state and status of a task which can enforces a rescheduling.
+//
+// Notes: make sure the clocks of aggregated-apiserver and scheduler are synchronized when using this API.
+type Reschedule struct {
```

Comment:
Does creating a `Reschedule` CR only execute it once? How do I determine if the `Reschedule` has already been executed?

If the `Reschedule` is executed, does it not have any status to indicate that?

Reply:
Sorry for missed your this comment!

Now, the complete CR is like this:

```yaml
apiVersion: apps.karmada.io/v1alpha1
kind: WorkloadRebalancer
metadata:
  name: demo
spec:
  ttlMinutesAfterFinished: 60
  workloads:
  - apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    name: demo-admin
  - apiVersion: apps/v1
    kind: Deployment
    name: demo-test-2
    namespace: default
status:
  lastUpdateTime: "2024-04-30T09:16:47Z"
  observedGeneration: 1
  observedWorkloads:
  - reason: ReferencedBindingNotFound
    result: Failed
    workload:
      apiVersion: rbac.authorization.k8s.io/v1
      kind: ClusterRole
      name: demo-admin
  - result: Successful
    workload:
      apiVersion: apps/v1
      kind: Deployment
      name: demo-test-2
      namespace: default
```

> Does creating a Reschedule CR only execute it once?

yes, execute it once with the result `Successful` or `Failed/ReferencedBindingNotFound`, individual failed workloads will be retried until they succeed.

this CR can also support `update` operation, implemented in #4860

> How do I determine if the Reschedule has already been executed?

see `status` field

> If the Reschedule is executed, does it not have any status to indicate that?

yes, it has

Reply:
hello, you may still confused about this proposal, so I made a thorough perfection of it.

now, you can review it again and feel free to ask any questions.